### ISSUE TITLE
Detect duplicate schedule names when cloning a schedule

### ISSUE THREAD

**Issue by benoit74** [openzim/zimfarm#844]
- Location: API / UI
 
## Problem
When cloning a schedule, if the new chosen name is already in used, this is only detected at the DB level, raising a 500 error.

We should detect this at least at the API level, and maybe also at the UI level (effort to be investigated).

## Reproducing steps
Clone a schedule and choose a name which is already in use.

**Comment by kelson42** [comment]
Duplicate of #740

**Comment by benoit74** [comment]
This is not really a duplicate, I was only speaking about duplicated names. Not about duplicated configuration as in #740. This issue is probably way easier to solve than #740 and would avoid 500 errors

**Comment by kelson42** [comment]
@benoit74 Today, I lost around 15 mn because of that issue. Considering we have a PR we should anyway be treated/improved/... and the impact is very "vicious" (old ZIM disappear from library.kiwix.org) Could you please fix this in priority?

**Comment by benoit74** [comment]
@kelson42 I don't think your problem is the same as this issue. This issue (and corresponding PR) is speaking about duplicated **schedule** name. You seem to have an issue about duplicate **ZIM** name. This is a distinct problem far more complex to solve (please open an issue if you confirm your problem is a different problem). As indicated here, this issue is only about better handling of duplicate schedule name which are currently raising 500 errors. And this is seemed not a very important thing since it is **very rare** to happen in our history (we are being alarmed on every 500 error since months now).

**Comment by benoit74** [comment]
@kelson42 I'm still waiting on your feedback about your recent issue. Can you clarify what made you loose 15 mn of your precious time so that we know what needs to be enhanced to not do it again? I'm quite sure your problem is duplicate ZIM name and not duplicate recipe name, but I would like to have your confirmation.

**Comment by kelson42** [comment]
@benoit74 You are 100% right, this is a duplicate ZIM metadata "Name". My bad.

**Comment by benoit74** [comment]
I isolated your specific issue here: https://github.com/openzim/zimfarm/issues/1037

**Comment by elfkuzco** [comment]
Is this not resolved by #862 ? Or is it the pending issue of not differentiating the exceptions?

**Comment by benoit74** [comment]
Yes it is going to be solved by #862. 

But we never achieved to merge this (not because it is complex, but because I was expecting Renaud to complete the effort, then Renaud handed over but I did not find time to complete the effort, and now the PR is so old it will not merged).

@elfkuzco can you please help finish this PR so that we can finally merge?

**Comment by elfkuzco** [comment]
Okay.

**Comment by elfkuzco** [comment]
Implemented this on #1245 on the `zimfarm_upgrade`
